lib/vmpooler/providers.rb

```diff
@@ -38,7 +40,7 @@ def load_from_gems(name = nil)
         # we don't exactly know if the provider name matches the main file name that should be loaded
         # so we use globs to get everything like the name
         # this could mean that vsphere5 and vsphere6 are loaded when only vsphere5 is used
-        Dir.glob(File.join(gem_path, "*#{name}*.rb")).each do |file|
+        Dir.glob(File.join(gem_path, "*#{name}*.rb")).sort.each do |file|
```

Comment:
This one is interesting:
```
lib/vmpooler/providers.rb:40:9: W: [Corrected] Lint/NonDeterministicRequireOrder: Sort files before requiring them.
        Dir.glob(File.join(gem_path, "*#{name}*.rb")).each do |file|
```

Reply:
I'd imagine it's so load order is consistent. It would probably make it harder to troubleshoot if the load order was inconsistent and a gem were part of the problem.